control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -1409,6 +1488,155 @@ func (r *HostedControlPlaneReconciler) reconcilePKI(ctx context.Context, hcp *hy
+	// crds.Cr
+	infraNamespace := hcp.Namespace
+	// infraUrl := r.RestConfig.APIPath
+	// infraCertificateAuthorityData := string(r.RestConfig.CAData)
+	// infraToken := "TODO"
+	infraKubeconfigSecretName := "tenant-controller-kubeconfig"
+
+	tenantNamespace := "openshift-cluster-csi-drivers"
+	// tenantUrl := r.TenantRestConfig.APIPath
+	// tenantCertificateAuthorityData := string(r.TenantRestConfig.CAData)
+	// tenantToken := "TODO"
+	// tenantKubeconfigSecretName := "infra-kubeconfig"
+	_, err := createOrUpdate(ctx, r.TenantCrcClient, kubevirt_assets.GenerateTenantNamespace(tenantNamespace), NoopReconcile)
+	if err != nil {
+		return err
+	}
+
+	for _, object := range kubevirt_assets.GenerateTenantNodeServiceAccountResources(tenantNamespace) {
+		_, err := createOrUpdate(ctx, r.TenantCrcClient, object, NoopReconcile)
+		if err != nil {
+			return err
+		}
+	}
+
+	infraServiceaccount, infraRole, infraRoleBinding := kubevirt_assets.GenerateInfraServiceAccountResources(infraNamespace)
+	_, err = createOrUpdate(ctx, r.InfraCrcClient, infraServiceaccount, NoopReconcile)
+	if err != nil {
+		return err
+	}
+	_, err = createOrUpdate(ctx, r.InfraCrcClient, infraRole, NoopReconcile)
+	if err != nil {
+		return err
+	}
+	_, err = createOrUpdate(ctx, r.InfraCrcClient, infraRoleBinding, NoopReconcile)
+	if err != nil {
+		return err
+	}
+
+	//TODO
+	// r.Log.Info("Updating scc")
+	// scc := &securityv1.SecurityContextConstraints{}
+	// err := r.InfraCrcClient.Get(ctx, crclient.ObjectKey{Name: "privileged"}, scc)
+	// if err != nil {
+	// 	return err
+	// }
+	// patch := client.MergeFrom(scc.DeepCopy())
+	// serviceAccountGlobalName := "system:serviceaccount:" + infraNamespace + ":kubevirt-csi"
+	// r.Log.Info("Patching scc for serviceaccount: ", serviceAccountGlobalName)
+	// scc.Users = append(scc.Users, serviceAccountGlobalName)
+	// err = r.InfraCrcClient.Patch(ctx, scc, patch)
+	// if err != nil {
+	// 	return err
+	// }
+
+	var infraReconciledServiceAccount *corev1.ServiceAccount = &corev1.ServiceAccount{}
+	err = r.Client.Get(ctx, client.ObjectKeyFromObject(infraServiceaccount), infraReconciledServiceAccount)
+	if err != nil {
+		return err
+	}
+
+	_, err = createOrUpdate(ctx, r.TenantCrcClient, kubevirt_assets.GenerateTenantConfigmap(tenantNamespace, infraNamespace), NoopReconcile)
+	if err != nil {
+		return err
+	}
+
+	_, err = createOrUpdate(ctx, r.TenantCrcClient, kubevirt_assets.GenerateDaemonset(tenantNamespace), NoopReconcile)
+	if err != nil {
+		return err
+	}
+
+	for _, object := range kubevirt_assets.GenerateTenantControllerServiceAccountResources(tenantNamespace) {
+		_, err = createOrUpdate(ctx, r.TenantCrcClient, object, NoopReconcile)
+		if err != nil {
+			return err
+		}
+	}
+
+	tenantControllerKubeconfigSecret := &corev1.Secret{}
+	tenantControllerCA := &corev1.Secret{}
+	pki.ReconcileServiceAccountKubeconfig(tenantControllerKubeconfigSecret, tenantControllerCA, hcp, tenantNamespace, "kubevirt-csi-controller-sa")
```

Comment:
this returns an error that needs to be handled. Right now this is silently failing with this error

```
{"level":"error","ts":"2022-08-31T20:20:09Z","msg":"Reconciler error","controller":"hostedcontrolplane","controllerGroup":"hypershift.openshift.io","controllerKind":"HostedControlPlane","hostedControlPlane":{"name":"dvossel-hyper1","namespace":"clusters-dvossel-hyper1"},"namespace":"clusters-dvossel-hyper1","name":"dvossel-hyper1","reconcileID":"bc50cff2-82f5-4be1-bfeb-ba71529b9423","error":"failed to update control plane: failed to reconcile kubevirt csi driver: failed to reconcile serviceaccount client cert: invalid CA signer secret  for cert(cn=system:serviceaccount:openshift-cluster-csi-drivers:kubevirt-csi-controller-sa,o=[system:serviceaccounts])","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/hypershift/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:273\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/hypershift/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:234"}

```

Reply:
this will fix it
```
diff --git a/control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go b/control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go
index 431eac8a..00cae89a 100644
--- a/control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go
+++ b/control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go
@@ -1574,14 +1574,20 @@ func (r *HostedControlPlaneReconciler) reconcileKubevirtCSIDriver(ctx context.Co
 		}
 	}
 
+	rootCA := manifests.RootCASecret(hcp.Namespace)
+	if err := r.Get(ctx, client.ObjectKeyFromObject(rootCA), rootCA); err != nil {
+		return fmt.Errorf("failed to get root ca cert secret: %w", err)
+	}
 	tenantControllerKubeconfigSecret := &corev1.Secret{}
-	tenantControllerCA := &corev1.Secret{}
-	pki.ReconcileServiceAccountKubeconfig(tenantControllerKubeconfigSecret, tenantControllerCA, hcp, tenantNamespace, "kubevirt-csi-controller-sa")
-	tenantControllerKubeconfig := tenantControllerKubeconfigSecret.Data[util.KubeconfigKey]
+	err = pki.ReconcileServiceAccountKubeconfig(tenantControllerKubeconfigSecret, rootCA, hcp, tenantNamespace, "kubevirt-csi-controller-sa")
+	if err != nil {
+		return err
+	}
+
+	tenantControllerKubeconfigSecret.Name = infraKubeconfigSecretName
+	tenantControllerKubeconfigSecret.Namespace = infraNamespace
 
-	// tenantControllerKubeconfig := kubevirt_assets.GenerateKubeconfig(tenantUrl, tenantNamespace, tenantCertificateAuthorityData, tenantToken)
-	infraKubeconfigSecret := kubevirt_assets.GenerateKubeconfigSecret(infraKubeconfigSecretName, infraNamespace, string(tenantControllerKubeconfig))
-	_, err = createOrUpdate(ctx, r.InfraCrcClient, infraKubeconfigSecret, NoopReconcile)
+	_, err = createOrUpdate(ctx, r.InfraCrcClient, tenantControllerKubeconfigSecret, NoopReconcile)
 	if err != nil {
 		return err
 	}

```